pkg/ttl/ttlworker/job.go

```diff
@@ -146,8 +146,9 @@ func (job *ttlJob) finish(se session.Session, now time.Time, summary *TTLSummary
 		}
 
 		return nil
-	}, session.TxnModeOptimistic)
+	}, session.TxnModePessimistic)
```

Comment:
Tough it can reduce the issue but cannot fully fix it because some other cases can also cause the sql execution failed, i.e network failure. Should we also finish the job first before we remove it from memory, if the finish fails, it will try it again next time.

https://github.com/pingcap/tidb/blob/1e24d39608214cc463c1de913e9a308be1c8193c/pkg/ttl/ttlworker/job_manager.go#L561-L562

Reply:
Done.